airbyte-integrations/postgres-destination/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java

```diff
@@ -0,0 +1,343 @@
+  static final String COLUMN_NAME = "data";
+
+  @Override
+  public DestinationConnectionSpecification spec() throws IOException {
+    // return a jsonschema representation of the spec for the integration.
+    final String resourceString = MoreResources.readResource("spec.json");
+    return Jsons.deserialize(resourceString, DestinationConnectionSpecification.class);
+  }
+
+  @Override
+  public StandardCheckConnectionOutput check(JsonNode config) {
+    try {
+      final BasicDataSource connectionPool = getConnectionPool(config);
+      DatabaseHelper.query(connectionPool, ctx -> ctx.execute(
+          "SELECT *\n"
+              + "FROM pg_catalog.pg_tables\n"
+              + "WHERE schemaname != 'pg_catalog' AND schemaname != 'information_schema';"));
+
+      connectionPool.close();
+    } catch (Exception e) {
+      // todo (cgardens) - better error messaging for common cases. e.g. wrong password.
+      return new StandardCheckConnectionOutput().withStatus(Status.FAILURE).withMessage(e.getMessage());
+    }
+
+    return new StandardCheckConnectionOutput().withStatus(Status.SUCCESS);
+  }
+
+  @Override
+  public StandardDiscoverSchemaOutput discover(JsonNode config) {
+    throw new RuntimeException("Not Implemented");
+  }
+
+  /**
+   * Strategy:
+   * <p>
+   * 1. Create a temporary table for each stream
+   * </p>
+   * <p>
+   * 2. Accumulate records in a buffer. One buffer per stream.
+   * </p>
+   * <p>
+   * 3. As records accumulate write them in batch to the database. We set a minimum numbers of records
+   * before writing to avoid wasteful record-wise writes.
+   * </p>
+   * <p>
+   * 4. Once all records have been written to buffer, flush the buffer and write any remaining records
+   * to the database (regardless of how few are left).
+   * </p>
+   * <p>
+   * 5. In a single transaction, delete the target tables if they exist and rename the temp tables to
+   * the final table name.
+   * </p>
+   *
+   * @param config - integration-specific configuration object as json. e.g. { "username": "airbyte",
+   *        "password": "super secure" }
+   * @param schema - schema of the incoming messages.
+   * @return consumer that writes singer messages to the database.
+   * @throws Exception - anything could happen!
+   */
+  @Override
+  public DestinationConsumer<SingerMessage> write(JsonNode config, Schema schema) throws Exception {
+    // connect to db.
+    final BasicDataSource connectionPool = getConnectionPool(config);
+    Map<String, WriteConfig> writeBuffers = new HashMap<>();
+
+    // create tmp tables if not exist
+    for (final Stream stream : schema.getStreams()) {
+      final String tableName = stream.getName();
+      final String tmpTableName = stream.getName() + "_" + Instant.now().toEpochMilli();
+      DatabaseHelper.query(connectionPool, ctx -> ctx.execute(String.format(
+          "CREATE EXTENSION IF NOT EXISTS \"uuid-ossp\";\n"
```

Comment:
Why do you need this?

Reply:
I am injecting our own id in the output as a uuid. in order to do that we need to make sure that the uuid lib is installed in the database. the alternative here is to just save the uuid as a string so that we don't need to install the lib.

Reply:
We should prefer inserting as a string so we aren't modifying the user's db in unexpected ways. I wouldn't expect any ETL tool to install a plugin in my database.

Reply:
Onboard with @jrhizor

Reply:
kk